src/glue/jobs/s3_to_json_s3.py

```diff
@@ -159,13 +161,14 @@ def update_json_schemas(s3_obj, archive_map, json_schemas):
                 json_schemas=json_schemas,
                 self_ref_schema_list=self_ref_schema_list,
             )
-            novel_schema = True
-            for cached_schema in json_schemas:
-                if all([cached_schema[k] == json_schema[k] for k in file_metadata]):
-                    novel_schema = False
-                    break
-            if novel_schema:
-                json_schemas.append(json_schema)
+            if json_schema["schema"] is not None:
+                novel_schema = True
+                for cached_schema in json_schemas:
+                    if cached_schema["schema"]["$id"] == json_schema["schema"]["$id"]:
+                        novel_schema = False
+                        break
+                if novel_schema:
+                    json_schemas.append(json_schema)
```

Comment:
Rather than match schemas using the keys in `file_metadata` (which doesn't actually make much sense... not sure why I did it this way in the first place), I compare the schemas directly by their `$id`.

Reply:
Ah, so before this change I was even adding files without schemas to `json_schemas` – which still doesn't make sense, but it explains why I was comparing the keys in `file_metadata`.

Reply:
I updated the docstring to describe this behavior.